examples/simple.py

```diff
@@ -13,15 +13,19 @@
 # function is used to actually generate the token
 @app.route('/login', methods=['POST'])
 def login():
-    username = request.json.get('username', None)
-    password = request.json.get('password', None)
-    if username != 'test' or password != 'test':
-        return jsonify({"msg": "Bad username or password"}), 401
-
-    # Identity can be any data that is json serializable
-    ret = {'access_token': create_access_token(identity=username)}
-    return jsonify(ret), 200
-
+    if request.is_json:
+        params = request.get_json()
+        if 'username' in params.keys() and 'password' in params.keys():
+            if params['username'] != 'test' or params['password'] != 'test':
+                return jsonify({"msg": "Bad username or password"}), 401
+        else:
+            return jsonify({"msg": "Missing auth parameters"}), 401
+
+        # Identity can be any data that is json serializable
+        ret = {'access_token': create_access_token(identity=params['username'])}
+        return jsonify(ret), 200
+    else:
+        return jsonify({"msg": "Missing auth"}), 401
```

Comment:
Just to avoid a bunch of nested if's, can you lift this to the top? Something like

```python
if not request.is_json:
    return jsonify({"msg": "Missing auth"}), 401

# rest of the code, but not nested in an if block any more
````